### ISSUE TITLE
Requires Hosted Key

### ISSUE THREAD

**Issue by eljobe** [keybase/keybase-issues#2492]
Several times in the web interface, I am presented with a dialog which presents me the the option to take some action `in the browser (unavailable for you)` which then shows the following pop-up.

![image](https://cloud.githubusercontent.com/assets/679924/17681231/5477b7fe-633c-11e6-9073-fde7d036287e.png)

The text describes my situation pretty clearly. "Keybase doesn't have a client-encrypted copy of my private key."  But, then it says something which is confusing to me. "If you change your mind, this feature will unlock."

I don't remember making a decision **not** to host a client encrypted copy of my private key on the keybase server.  And even if I did change my mind, I'm not sure what actions I should take to make the fact known to keybase.

I've looked into the online help for the keybase command. I assume there should be a way to give a client-encrypted copy of my private key to the keybase server, but I can't figure out the right command. It **might** be `keybase pgp import` but that command just hangs. It's possible it is waiting on my to paste input into stdin, but the documentation doesn't indicate that, and I'm nervous about just pasting the contents of my private GPG key into that command without understanding what it does.

Sorry to file a bug for what is, in fact, a support request, but I couldn't figure out another place to ask for support. I don't see a user forum, and searching the internet for the message from the UI didn't come up with anything helpful. Please help.

**Comment by zQueal** [comment]
> I don't remember making a decision not to host a client encrypted copy of my private key on the keybase server. 

If you generated your key with Keybase, then you were given a choice to host or to not host your private key with Keybase. If this is the case, then you must have chosen the latter. If you didn't generate with Keybase then you must have only imported your public key.

You should be able to import your private key by doing;

    keybase pgp import -i private.key # import private key from backup
    keybase push -s # push secret key to Keybase

**Comment by eljobe** [comment]
@zQueal thanks for the quick response.

I did not generate my key with Keybase. I already had a gpg key that I was using.  I initially used:
`keybase gpg select` to select that key as my gpg key for use with Keybase.

Now, when I attempt to follow your instructions this is what I get:
```bash
$> gpg --export-secret-key -a pepper > ~/.gnupg/private-keys-v1.d/private.key
$> keybase gpg import -i ~/.gnupg/private-keys-v1.d/private.key
(Keybase UI Prompts me to enter my passphrase, which I do.)
▶ ERROR This key is already active. You can't add it twice. (error 923)
```

It seems that because I have already imported the public key into keybase, I'm not allowed to import the private key now.

**Comment by eljobe** [comment]
I'm wondering if I am making a mistake as a user or if I've just run up against some edge-case that is not working as expected by the development team.

**Comment by zQueal** [comment]
That's pretty interesting. I would expect that to work in 99% of cases, but I guess not.

Looks like #2050 is the same issue. My only guess would be to agree with him and say that Keybase is expecting you to have no private key locally, but you actually do (just not remotely).

**Comment by eljobe** [comment]
After reading [some of the code](https://github.com/keybase/client/blob/master/go/client/cmd_pgp_import.go#L37), it looks like I should have run `kebase gpg import --push-secret -i ~/.gnupg/private-keys-v1.d/private.key` instead of `keybase gpg select` when I was first picking my old gpg key to use with keybase.

Now, since the `keybase gpg import --push-secret` command is erroring out before it can even add my private key to keybase, I don't think it is getting to the part of the code that would make the API call to push the encrypted secret key to the server.

What are my options? Should I click this option in the webapp?
![image](https://cloud.githubusercontent.com/assets/679924/17817557/00573f98-6637-11e6-926b-e0852280c472.png)

It seems drastic. I guess I could try to fork the client and run just the part that pushes my encrypted private key to the server. But, that seems risky.

BTW, I find it pretty counter intuitive to run a command like this one `keybase pgp help import` to get the help message for that subcommand to print to the console. I think it would be cool if that could be `keybase help gpg import` instead. That's what I typed several times before reading the code.

**Comment by zQueal** [comment]
These are all some really good questions. One's I really can't seem to answer. Maybe @maxtaco can shed some light here.

One thing you could do (hopefully), is since you don't have your private key in the online keyring you should be able to add it via copy-paste (via somewhere in the online interface) and then do `keybase pgp pull` to pull it into your local Keybase keyring.

We might be missing something simple, but there doesn't seem to be a graceful way to accomplish this with ease if you missed it the first time.

**Comment by eljobe** [comment]
@maxtaco Any news from your investigation?

**Comment by maxtaco** [comment]
@eljobe Can you try it from the Web site?  Click on the tooltip next to the PGP fingerprint on your profile and click on "Host an encrypted copy of my private key"

I'm guessing there's a bug in the CLI for this feature, but very few have hit it, and I haven't had a chance to try to repro it.  I can file a ticket though.

**Comment by eljobe** [comment]
SUCCESS!  But, I did end up with some further confusion along the way, and wanted to make sure you were aware of the areas which tripped me up, so that you can make them easier for "the rest of us."

## UX Concerns

I would never have figured out that I wanted to click "edit" next to my private key to attempt to add an encrypted version of the private key. This is not very discoverable.

## Passphrase Error is Ambiguous

Maybe this is security by obscurity, but when I'm trying to encrypt my private key, I have to provide two passphrases. (1) The passphrase used to lock my private key in GPG. (2) My keybase passphrase. But, when I click Encrypt, I end up with this error message.

![image](https://cloud.githubusercontent.com/assets/679924/17978430/f2e27c68-6aed-11e6-94b6-2c43bf42a845.png)

It isn't clear which passphrase check failed. Also, wasn't sure if the `Keybase passphrase` I was supposed to enter was the password that I used to login to the website or my paper key. I later figured out that you were looking for the password used to login to the website.

## Bug on subsequent attempts to use the `Private key encrypt and upload` dialog

### Steps to reproduce
1. Open the dialog via "edit" > "Host an encrypted copy of my private key"
2. Paste in your private key
3. Enter the passphrase on the private key
4. Enter an *incorrect* Keybase passphrase
5. Click "Encrypt"
6. Observe that you get the error mentioned above
7. Close the dialog box with the little "x" button
8. Repeat Step 1
9. Repeat Step 2

### Observed Behavior
After pasting in your private key for the second time, and error is shown (something to the effect of "private key has a passphrase but you didn't supply one.") and the text field I was able to use in step 3 above is not shown.

### Expected Behavior
The dialog box should work the same the second time as it did the first time I used it.

Workaround:
I found that if I would use the browser to refresh the whole page, I could get it back into a good state.

If you'd like I can file a separate issue just to track this subtle UI bug.

**Comment by REBELinBLUE** [comment]
I've been having exactly the same issue and it was only because of @eljobe post there that I even noticed what the problem was

**Comment by runn1ng** [comment]
Well, I got to the same point, I have no idea what to do.

I did `keybase pgp select`. I will "change my mind"? How do I change my mind? I cannot find any of the settings you describe in this thread

**Comment by fabianhjr** [comment]
@runn1ng 
```
keybase pgp export -s | keybase pgp import --push-secret
```

Same error
